### PR TITLE
Clarify initial_master_nodes must match node.name

### PR DESCRIPTION
**Pull request by DaveCTurner**
... and emphasize that this includes any trailing qualifiers.

### CODE REVIEW

File: docs/reference/modules/discovery/bootstrapping.asciidoc [hunk 1]
```diff
@@ -56,6 +56,28 @@ cluster.initial_master_nodes:
   - master-node-hostname
 --------------------------------------------------
 
+[NOTE]
+==================================================
+The node names used in this list must exactly match the `node.name` properties
+of the nodes. If your node names are a fully-qualified domain name such as
```
Comment (bleskes) [inline docs/reference/modules/discovery/bootstrapping.asciidoc]: Wdyt about:

By default the `node.name` setting is resolved to the machine hostname. If the host name is fully-qualified domain name such as ...

File: docs/reference/modules/discovery/bootstrapping.asciidoc [hunk 2]
```diff
@@ -56,6 +56,28 @@ cluster.initial_master_nodes:
...
+`master-a.example.com` then you must include the fully-qualified domain name in
+the `cluster.initial_master_nodes` list too; conversely if your node names are
+bare hostnames (without the `.example.com` suffix) then you must use bare
+hostnames in the `cluster.initial_master_nodes` list too. If you use a mix of
```
Comment (bleskes) [inline docs/reference/modules/discovery/bootstrapping.asciidoc]: > If you use a mix of fully-qualifed and bare hostnames then the cluster will not properly form and
you will see log messages like the following, indicating that the names do not match exactly.

How about: if the node names and the `cluster.initial_master_nodes` list don't much exactly, you are likely to see log message like the following:

**Review by bleskes — COMMENTED** [review]
Thx David. I left some suggestions.

File: docs/reference/modules/discovery/bootstrapping.asciidoc (hunk 1 above)
Comment (DaveCTurner) [reply docs/reference/modules/discovery/bootstrapping.asciidoc]: Sure; we mention the default a few lines above but it seems fine to mention it again. I did this in eacb7640de0.

File: docs/reference/modules/discovery/bootstrapping.asciidoc (hunk 2 above)
Comment (DaveCTurner) [reply docs/reference/modules/discovery/bootstrapping.asciidoc]: I want to keep the bit about a "mix of fully-qualifed and bare hostnames" but I added some more general words in eacb7640de0 too.

**Review by DaveCTurner — COMMENTED** [review]
Thanks @bleskes, suggestions actioned.

**Review by bleskes — APPROVED** [review]
LGTM

**Pull request merged**